### PR TITLE
Added social accounts verification logic

### LINKED ISSUE #27: Influencer Social Handles need Verification before allowing to signUp
## Description
Currently, there is no way to verify if the social profile influencer has given belongs to him/her. 

## Expectation

- [ ] There should be a way to verify that the particular social account claimed by the influencer belong to him/her only.
- [ ] Influencers shouldn't be allowed to signup unless they verify their mentioned social profile. 

Screenshot 2024-06-11 at 11 04 55 PM

### PR DESCRIPTION
**Pull request by Asymtode712**
## Description

Fixes #27 

Added the total logic for social accounts verification by creating `socialsverification.js` 

@RohitGupta1237 You have to generate the API tokens for each Social on your part, I would provide the steps for doing it, so that it would be easy for you

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Never Ever add `node_modules` in your prs!

**Comment by Asymtode712** [comment]
@Kajol-Kumari So sorry, actually the `.gitignore` file wasn't properly configured

**Comment by Kajol-Kumari** [comment]
@Asymtode712 i am still seeing `node_module` folder in this pr, either clean that up or please raise a new pr which i can review

**Comment by Asymtode712** [comment]
@Kajol-Kumari Done....Sorry once again😓🙏🏼

File: backend/.gitignore [hunk 1]
```diff
@@ -1,2 +1,2 @@
-node_modules/
```
Comment (Kajol-Kumari) [inline backend/.gitignore:1]: please retain this as well as it seems to be working for few folks.

File: backend/controllers/socialsverification.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
+const fetch = require('node-fetch');
+const oauth = require('oauth');
+
+// Instagram Verification
+const verifyInstagram = async (instagramHandle) => {
+  const instagramApiUrl = 'https://graph.instagram.com/me';
```
Comment (Kajol-Kumari) [inline backend/controllers/socialsverification.js:6]: please keep this either in .env config or in a constant file

File: backend/controllers/socialsverification.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    console.error('Error verifying Instagram profile:', error);
+  }
+
+  return false;
+};
+
+// YouTube Verification
+const verifyYouTube = async (youtubeChannel) => {
+  const googleApiKey = process.env.GOOGLE_API_KEY; 
+  const youtubeApiUrl = 'https://www.googleapis.com/youtube/v3/channels';
```
Comment (Kajol-Kumari) [inline backend/controllers/socialsverification.js:27]: please keep this either in .env config or in a constant file

File: backend/controllers/socialsverification.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  } catch (error) {
+    console.error('Error verifying YouTube channel:', error);
+  }
+
+  return false;
+};
+
+// Facebook Verification
+const verifyFacebook = async (facebookProfile) => {
+  const facebookApiUrl = 'https://graph.facebook.com/v15.0/me';
```
Comment (Kajol-Kumari) [inline backend/controllers/socialsverification.js:45]: please keep this either in .env config or in a constant file

File: backend/controllers/socialsverification.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+// Twitter Verification
+const verifyTwitter = async (twitterHandle) => {
+  const twitterConsumerKey = process.env.TWITTER_CONSUMER_KEY; 
+  const twitterConsumerSecret = process.env.TWITTER_CONSUMER_SECRET; 
+  const twitterAccessToken = process.env.TWITTER_ACCESS_TOKEN; 
+  const twitterAccessTokenSecret = process.env.TWITTER_ACCESS_TOKEN_SECRET; 
+
+  const oauthClient = new oauth.OAuth(
+    'https://api.twitter.com/oauth/request_token',
```
Comment (Kajol-Kumari) [inline backend/controllers/socialsverification.js:71]: please keep these 2 endpoints either in .env config or in a constant file

File: package.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (Kajol-Kumari) [inline package.json:1]: what's the purpose of this new `package.json` file? why can't we have the dependencies in `backend/package.json` file?

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
The package-lock.json file should also not be there as all the backend required dependencies should be under `backend/package.json` file

**Comment by RohitGupta1237** [comment]
@Asymtode712 The logic is not working in the way we wanted it to  work.

**Pull request closed without merging**